dataland-backend/src/main/kotlin/org/dataland/datalandbackend/configurations/SwaggerUiConfig.kt

```diff
@@ -0,0 +1,23 @@
+package org.dataland.datalandbackend.configurations
+
+import org.springdoc.core.models.GroupedOpenApi
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+
+/**
+ * This class is used to configure the CSP for the Swagger-UI as the default content does not allow loading anything
+ */
+
+@Configuration
+class SwaggerUiConfig {
+
+    /**
+     * This method returns all public endpoints to be displayed in the swagger ui
+     */
+    @Bean
+    fun publicApi(): GroupedOpenApi? {
+        return GroupedOpenApi.builder().group("public").pathsToExclude("/internal/**").addOpenApiCustomizer(
+            DataTypeSchemaCustomizer(),
+        ).build()
```

Comment:
Will
`api/v3/api-docs` still contain the internal endpoints?

Reply:
> It does. I assume it is not supposed to.

Yes - it is not supposed to.

Reply:
As just learned: The Link to the OpenApi Definitions file on the swaggger-ui does not containt the internal links. In this case I feel that this is a nice solution. So no further todos - i will resolve this conversation.